sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_intel_matrix.asciidoc

```diff
@@ -203,6 +203,147 @@ In the case of `ext_intel_packed` matrix memory layout, `row` and
+void joint_matrix_load_checked(Group g,
+   joint_matrix<Group, T1, use::accumulator, Rows, Cols, layout::dynamic> &res,
+   ext::oneapi::experimental::annotated_ptr<T2, PropertyListT> base_src,
+   size_t Stride, layout Layout, size_t Height, size_t Width, size_t
+   coordX, size_t coordY);
+
+// Only available when Layout != layout::dynamic
+// and when std::is_same_v<T1, std::remove_const_t<T2>>
+template <typename Group, typename T1, typename T2, size_t Rows,
+          size_t Cols, use Use, layout Layout, typename PropertyListT>
+void joint_matrix_load_checked(Group g,
+   joint_matrix<Group, T1, Use, Rows, Cols, Layout> &res,
+   ext::oneapi::experimental::annotated_ptr<T2, PropertyListT> base_src,
+   size_t Stride, size_t Height, size_t Width, size_t coordX, size_t coordY);
+
+template <typename Group, typename T, size_t Rows, size_t Cols,
+          access::address_space Space, access::decorated IsDecorated>
+void joint_matrix_store_checked(Group g,
+   const joint_matrix<Group, T, use::accumulator, Rows, Cols, layout::dynamic> &res, 
+   multi_ptr<T, Space, IsDecorated> base_dest, size_t Stride, layout Layout,
+   size_t Height, size_t Width, size_t coordX, size_t coordY);
+
+// Only available when Layout != layout::dynamic
+template <typename Group, typename T, size_t Rows, size_t Cols,
+          use Use, layout Layout, access::address_space Space,
+          access::decorated IsDecorated>
+void joint_matrix_store_checked(Group g,
+   const joint_matrix<Group, T, Use, Rows, Cols, Layout> &res,
+   multi_ptr<T, Space, IsDecorated> base_dest, size_t Stride,
+   size_t Height, size_t Width, size_t coordX, size_t coordY);
+
+template <typename Group, typename T, size_t Rows, size_t Cols,
+          typename PropertyListT>
+void joint_matrix_store_checked(Group g,
+   const joint_matrix<Group, T, use::accumulator, Rows, Cols, layout::dynamic> &res, 
+   ext::oneapi::experimental::annotated_ptr<T, PropertyListT> base_dest,
+   size_t Stride, layout Layout, size_t Height, size_t Width,
+   size_t coordX, size_t coordY);
+
+// Only available when Layout != layout::dynamic
+template <typename Group, typename T, size_t Rows, size_t Cols,
+          use Use, layout Layout, typename PropertyListT>
+void joint_matrix_store_checked(Group g,
+   const joint_matrix<Group, T, Use, Rows, Cols, Layout> &res,
+   ext::oneapi::experimental::annotated_ptr<T2, PropertyListT> base_dest,
+   size_t Stride, size_t Height, size_t Width, size_t coordX, size_t coordY);
+
+} // namespace sycl::ext::intel::experimental::matrix
+```
+
+The property list associated with the `annotated_ptr` argument
+represents the compile-time constant properties for cache control included
+in the SYCL extenion
+link:../../proposed/sycl_ext_intel_cache_controls.asciidoc[sycl_ext_intel_cache_controls],
+and for the alignment property defined in the SYCL extension
+link:../../experimental/sycl_ext_oneapi_kernel_arg_properties.asciidoc[sycl_ext_oneapi_kernel_arg_properties].
```

Comment:
I will remove alignment from the checked APIs because only cache control is what is being considered for implementation so far.
Alignments will be handled by the query like you suggested in your last comment. So users must make sure their pointers are 4 bytes aligned. Otherwise, results are undefined.